### DISCUSSION TITLE
Propose extensions for new users in open-wc templates

### DISCUSSION THREAD

**Discussion by jlengrand** [open-wc/open-wc#2245] [Ideas]
Hey there, 

I wanna check the idea past you before proposing a PR :).

You folks [recommend vscode and given extensions](https://open-wc.org/guides/tools/ide/) in your documentation.

There is a way to offer to install recommendations when you create / clone a project and the user doesn't have them installed yet in vscode. [See this page](https://code.visualstudio.com/docs/editor/extension-marketplace#_workspace-recommended-extensions)

I would like to recommend creating the extension.json file in the .code folder. This is how it will look like.

* In the template, add a `.code/extensions.json` file that looks like this : 

```
{
	"recommendations": [
		"esbenp.prettier-vscode", 
		"runem.lit-plugin",
		"dbaeumer.vscode-eslint"
 // Add typescript for TS projects?
	],
	"unwantedRecommendations": [

	]
}
```

* When the user creates a new project and extensions are not installed yet, he is prompted with a pop up like this : 

![image](https://user-images.githubusercontent.com/921666/125419192-2282f72e-b95b-4681-9264-7394145789d2.png)

* This shows recommended extensions and allows me to install them at the press of a single button : 

![image](https://user-images.githubusercontent.com/921666/125419264-24bdfb9a-cfaf-42a3-b330-491d4d6702c0.png)

What do you think ?

**Comment by jlengrand** [comment]
Hum, I'm looking into the create repo, if you have any pointers as to where the .code folder and the extensions.json files should be placed in the generators that'd actually be helpful

**Reply by daKmoR** [reply]
you can probably add it to the common repo helpers
https://github.com/open-wc/create/tree/master/src/generators/common-repo/templates/static

**Reply by daKmoR** [reply]
hmmmm `.` files are always a little tricky... might need to add some special copying 😅

just a hunch - be sure to test it out

**Reply by jlengrand** [reply]
Didyou try it? I ve made a PR for it yesterday night 🙂
